graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/IncrementModelBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.graphqlfirst.processor;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import graphql.Scalars;
+import graphql.language.Directive;
+import graphql.language.InputValueDefinition;
+import graphql.language.Type;
+import graphql.language.TypeName;
+import graphql.schema.idl.TypeUtil;
+import io.stargate.graphql.schema.graphqlfirst.util.TypeHelper;
+import io.stargate.graphql.schema.scalars.CqlScalar;
+import java.util.*;
+
+public class IncrementModelBuilder extends ArgumentDirectiveModelBuilderBase<IncrementModel> {
+
+  // The types that we allow for an argument that represents an increment to a counter:
+  private static final List<String> COUNTER_INCREMENT_TYPES =
+      ImmutableList.of(
+          Scalars.GraphQLInt.getName(),
+          CqlScalar.BIGINT.getGraphqlType().getName(),
+          CqlScalar.COUNTER.getGraphqlType().getName());
```

Comment:
I've added `Counter`, it feels natural to use the same type as the field, and this didn't require any additional code.